### DISCUSSION TITLE
What is the purpose of the GPIO2 connected to ground in PCB v1.1?

### DISCUSSION THREAD

**Discussion by walklinewang** [hrvach/deskhop#126] [Q&A] [answered]
I found this when I was redrawing the schematic, U1's GPIO2 was connected to ground but U2's wasn't. I'm wondering if this is used to identify the master device or has some other purpose, but I didn't find any GPIO2 related definitions or IO manipulation code in the source code.

![image](https://github.com/user-attachments/assets/fba238a3-6619-4863-8db7-bd58a81b1351)

**Comment by hrvach** [comment, marked as answer]
Great catch, and you are right - the idea was to provide a way to identify which board the code was running on, but that would mean handling old boards differently. In the meantime I managed to design a detection routine for the existing boards by using internal pull-up/downs, so it's proooobably going to be deprecated.

**Comment by walklinewang** [comment]
Thanks, I know how to deal with it now.
